### PR TITLE
(pillan) add prometheus-snmp-exporter

### PR DESCRIPTION
**Pull request by jhoblitt**
Depends on https://github.com/lsst-it/k8s-cookbook/pull/281

Note that snmp-exporter-pre will require some modifications to work with multiple clusters. That work is planned to be PR'd separately.

### CODE REVIEW

File: fleet/lib/snmp-exporter-pre/templates/externalsecret-snmp-community.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  name: snmp-config
+spec:
+  secretStoreRef:
+    kind: ClusterSecretStore
+    name: onepassword
+  data:
+    - secretKey: snmpRubinCommunity
+      remoteRef:
+        key: snmp community string
+        property: credential
```
Comment (fbegyn) [inline fleet/lib/snmp-exporter-pre/templates/externalsecret-snmp-community.yaml:14]: There is also a `snmpLsstCommunity` in the config file, so it needs to be exposed as well in the externalSecret

File: fleet/lib/snmp-exporter-pre/files/snmp.yml [hunk 2]
```diff
@@ -0,0 +1,2055 @@
...
+    security_level: noAuthNoPriv
+    auth_protocol: MD5
+    priv_protocol: DES
+    version: 1
+  lsst_v2:
+    community: '{{.snmpLsstCommunity}}'
+    security_level: noAuthNoPriv
+    auth_protocol: MD5
+    priv_protocol: DES
+    version: 2
```
Comment (fbegyn) [inline fleet/lib/snmp-exporter-pre/files/snmp.yml:14]: The `snmpLsstCommunity`  here is not defined in the externalSecret.

File: fleet/lib/snmp-exporter-pre/files/sd/snmp-network.json
Comment (fbegyn) [inline fleet/lib/snmp-exporter-pre/files/sd/snmp-network.json:1]: Ideally these JSON files would be replaced by a Netbox/SD that holds the network devices.

**Review by fbegyn — COMMENTED** [review]
Aside from the externalSecret, just a small remark on the sourcing of the network devices (which is fine currently until a better solution is found).

File: fleet/lib/snmp-exporter-pre/templates/externalsecret-snmp-community.yaml (hunk 1 above)
Comment (jhoblitt) [reply fleet/lib/snmp-exporter-pre/templates/externalsecret-snmp-community.yaml:14]: Interpolating this value was intentionally omitted as that community string isn't used in Tucson.

File: fleet/lib/snmp-exporter-pre/files/snmp.yml (hunk 2 above)
Comment (jhoblitt) [reply fleet/lib/snmp-exporter-pre/files/snmp.yml:14]: As mentioned https://github.com/lsst-it/k8s-cookbook/pull/280#discussion_r1534779064, this is intentional and still results in a working configuration.

File: fleet/lib/snmp-exporter-pre/templates/externalsecret-snmp-community.yaml (hunk 1 above)
Comment (jhoblitt) [reply fleet/lib/snmp-exporter-pre/templates/externalsecret-snmp-community.yaml:14]: In another branch, I have work to to template the `externalsecrets` resources with helm values.

File: fleet/lib/snmp-exporter-pre/files/sd/snmp-network.json
Comment (jhoblitt) [reply fleet/lib/snmp-exporter-pre/files/sd/snmp-network.json:1]: Agreed but that's probably not going to happen for months.

File: fleet/lib/snmp-exporter-pre/templates/externalsecret-snmp-community.yaml (hunk 1 above)
Comment (jhoblitt) [reply fleet/lib/snmp-exporter-pre/templates/externalsecret-snmp-community.yaml:14]: I've put the multi-site support into https://github.com/lsst-it/k8s-cookbook/pull/285

**Review by fbegyn — APPROVED** [review]
LGTM, comments resolved.

**Pull request merged**